### PR TITLE
fix: LcVarsIterMut::size_hint off-by-one and potential underflow

### PR DESCRIPTION
**Pull request by Galoretka**
Use Windows::len() directly as the exact number of remaining windows instead of subtracting 1.
The previous implementation self.offsets.len() - 1 underestimated the remaining items and could underflow when the iterator was empty (panic in debug, wrap in release).
This aligns size_hint with the semantics of core::slice::Windows, ensuring accurate iteration hints and avoiding edge-case failures.

### CODE REVIEW

**Comment by Pratyush** [comment]
Can you add a test which demonstrates this is an issue? `self.offsets.len()` is always 1 at least.

**Comment by Galoretka** [comment]
> Can you add a test which demonstrates this is an issue? `self.offsets.len()` is always 1 at least.

Yes, done

File: relations/src/gr1cs/lc_map.rs [hunk 1]
```diff
@@ -305,7 +305,7 @@ impl<'a> Iterator for LcVarsIterMut<'a> {
 
     #[inline]
     fn size_hint(&self) -> (usize, Option<usize>) {
-        let len = self.offsets.len() - 1;
+        let len = self.offsets.len();
```
[the remark this answers is not in the document]
Comment (Pratyush) [reply relations/src/gr1cs/lc_map.rs:308]: hm could you please check this case as well? Thank you!

[the remark this answers is not in the document]
Comment (Galoretka) [reply relations/src/gr1cs/lc_map.rs:308]: > hm could you please check this case as well? Thank you!

In LcVarsIterMut the field named offsets is actually a Windows<'_, usize> iterator (from offsets.windows(2)), not the raw offsets slice. For Windows, .len() already equals “how many windows are left.” So size_hint should use (len, Some(len)) as-is.
- If we subtracted 1 here, we’d undercount and also break the empty case; the new test covers that and would fail.
- In the parallel path we do use the raw slice, so we correctly use offsets.len().saturating_sub(1). Both paths are consistent.

So the current size_hint is correct, and the test is right.

**Review by Pratyush — APPROVED** [review]
Thank you!
